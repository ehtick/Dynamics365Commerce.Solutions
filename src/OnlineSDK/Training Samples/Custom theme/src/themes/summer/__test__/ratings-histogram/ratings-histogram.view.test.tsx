/*!
 * Copyright (c) Microsoft Corporation.
 * All rights reserved. See LICENSE in the project root for license information.
 */

/* eslint-disable no-duplicate-imports */
import { buildMockModuleProps } from '@msdyn365-commerce/core';
import { render } from 'enzyme';
import * as React from 'react';

import RatingsHistogramView from '../../views/ratings-histogram.view';

const mockData = {
    ratingsSummary: {},
    ratingsReviewsState: {}
};

const mockString = 'stars';
const createItem = (stars: number) => {
    return {
        starLabel: (
            <div>
                {stars} {mockString}
            </div>
        ),
        bar: (
            <div>
                {stars} {mockString}
            </div>
        ),
        percentage: (
            <div>
                {stars} {mockString}
            </div>
        ),
        buttonProps: { className: 'foo' }
    };
};

const items = [createItem(5), createItem(4), createItem(3), createItem(2), createItem(1)];

let mockProps = {};

describe('Write Review unit tests - View', () => {
    it('renders correctly', () => {
        const moduleProps = buildMockModuleProps({}, {});
        const selectFilterBy = jest.fn();
        mockProps = {
            ...moduleProps,
            className: 'className',
            data: mockData,
            moduleProps: {
                moduleProps,
                className: 'className'
            },
            histogramItems: items,
            averageRating: '<div>ratings</div>',
            heading: '<h2>heading</h2>',
            averageNumber: '<p>3</p>',
            totalReviews: '<p>30</p>',
            ratingsSummaryProps: { className: 'ms-ratings-histogram__summary' },
            histogramProps: { className: 'ms-ratings-histogram__histogram' },
            callbacks: {
                selectFilterBy
            }
        };

        // @ts-expect-error
        const component = render(<RatingsHistogramView {...mockProps} />);
        expect(component).toMatchSnapshot();
    });
});
