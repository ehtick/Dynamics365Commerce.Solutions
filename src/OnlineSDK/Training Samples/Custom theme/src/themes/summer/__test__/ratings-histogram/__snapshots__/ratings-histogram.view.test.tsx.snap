// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Write Review unit tests - View renders correctly 1`] = `
<div
  class="className"
  data-m-t="mock-type"
  id=""
>
  &lt;h2&gt;heading&lt;/h2&gt;
  <div
    class="ms-ratings-histogram__histogram"
  >
    <div
      class="foo"
    >
      <div>
        5 stars
      </div>
      <div>
        5 stars
      </div>
      <div>
        5 stars
      </div>
    </div>
    <div
      class="foo"
    >
      <div>
        4 stars
      </div>
      <div>
        4 stars
      </div>
      <div>
        4 stars
      </div>
    </div>
    <div
      class="foo"
    >
      <div>
        3 stars
      </div>
      <div>
        3 stars
      </div>
      <div>
        3 stars
      </div>
    </div>
    <div
      class="foo"
    >
      <div>
        2 stars
      </div>
      <div>
        2 stars
      </div>
      <div>
        2 stars
      </div>
    </div>
    <div
      class="foo"
    >
      <div>
        1 stars
      </div>
      <div>
        1 stars
      </div>
      <div>
        1 stars
      </div>
    </div>
  </div>
</div>
`;
